File: app/src/main/java/com/example/administrator/encyclopedia/bean/ZhouGongBean.java
package com.example.administrator.encyclopedia.bean;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import org.json.JSONException;
import org.json.JSONObject;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

/**
 * 项目名：   Encyclopedia
 * 包名 ：    com.example.administrator.encyclopedia.bean
 * 文件名：   ZhouGongBean
 * 创建者:    Nixo
 * 创建时间：  2018/5/12/012-18:47
 * 描述：
 */

public class ZhouGongBean {

    /**
     * reason : successed
     * result : [{"id":"ec196627ef7a3d6c394222ac7a199c3d","title":"火车票","des":"梦见火车票，吉兆，预示着通过自己的努力已经成功。"}]
     * error_code : 0
     */

    private String reason;
    private int error_code;
    private List<ResultBean> result;

    public static ZhouGongBean objectFromData(String str) {

        return new Gson().fromJson(str, ZhouGongBean.class);
    }

    public static ZhouGongBean objectFromData(String str, String key) {

        try {
            JSONObject jsonObject = new JSONObject(str);

            return new Gson().fromJson(jsonObject.getString(str), ZhouGongBean.class);
        } catch (JSONException e) {
            e.printStackTrace();
        }

        return null;
    }

    public static List<ZhouGongBean> arrayZhouGongBeanFromData(String str) {

        Type listType = new TypeToken<ArrayList<ZhouGongBean>>() {
        }.getType();

        return new Gson().fromJson(str, listType);
    }

    public static List<ZhouGongBean> arrayZhouGongBeanFromData(String str, String key) {

        try {
            JSONObject jsonObject = new JSONObject(str);
            Type listType = new TypeToken<ArrayList<ZhouGongBean>>() {
            }.getType();

            return new Gson().fromJson(jsonObject.getString(str), listType);

        } catch (JSONException e) {
            e.printStackTrace();
        }

        return new ArrayList();


    }

    public String getReason() {
        return reason;
    }

    public void setReason(String reason) {
        this.reason = reason;
    }

    public int getError_code() {
        return error_code;
    }

    public void setError_code(int error_code) {
        this.error_code = error_code;
    }

    public List<ResultBean> getResult() {
        return result;
    }

    public void setResult(List<ResultBean> result) {
        this.result = result;
    }

    public static class ResultBean {
        /**
         * id : ec196627ef7a3d6c394222ac7a199c3d
         * title : 火车票
         * des : 梦见火车票，吉兆，预示着通过自己的努力已经成功。
         */

        private String id;
        private String title;
        private String des;

        public static ResultBean objectFromData(String str) {

            return new Gson().fromJson(str, ResultBean.class);
        }

        public static ResultBean objectFromData(String str, String key) {

            try {
                JSONObject jsonObject = new JSONObject(str);

                return new Gson().fromJson(jsonObject.getString(str), ResultBean.class);
            } catch (JSONException e) {
                e.printStackTrace();
            }

            return null;
        }

        public static List<ResultBean> arrayResultBeanFromData(String str) {

            Type listType = new TypeToken<ArrayList<ResultBean>>() {
            }.getType();

            return new Gson().fromJson(str, listType);
        }

        public static List<ResultBean> arrayResultBeanFromData(String str, String key) {

            try {
                JSONObject jsonObject = new JSONObject(str);
                Type listType = new TypeToken<ArrayList<ResultBean>>() {
                }.getType();

                return new Gson().fromJson(jsonObject.getString(str), listType);

            } catch (JSONException e) {
                e.printStackTrace();
            }

            return new ArrayList();


        }

        public String getId() {
            return id;
        }

        public void setId(String id) {
            this.id = id;
        }

        public String getTitle() {
            return title;
        }

        public void setTitle(String title) {
            this.title = title;
        }

        public String getDes() {
            return des;
        }

        public void setDes(String des) {
            this.des = des;
        }
    }
}

File: app/src/main/java/com/example/administrator/encyclopediamvp/Bean/ZhouGongBean.kt
package com.example.administrator.encyclopediamvp.Bean

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

import org.json.JSONException
import org.json.JSONObject

import java.lang.reflect.Type
import java.util.ArrayList

/**
 * 项目名：   Encyclopedia
 * 包名 ：    com.example.administrator.encyclopedia.bean
 * 文件名：   ZhouGongBean
 * 创建者:    Nixo
 * 创建时间：  2018/5/12/012-18:47
 * 描述：
 */

class ZhouGongBean {

    /**
     * reason : successed
     * result : [{"id":"ec196627ef7a3d6c394222ac7a199c3d","title":"火车票","des":"梦见火车票，吉兆，预示着通过自己的努力已经成功。"}]
     * error_code : 0
     */

    var reason: String? = null
    var error_code: Int = 0
    var result: List<ResultBean>? = null

    class ResultBean {
        /**
         * id : ec196627ef7a3d6c394222ac7a199c3d
         * title : 火车票
         * des : 梦见火车票，吉兆，预示着通过自己的努力已经成功。
         */

        var id: String? = null
        var title: String? = null
        var des: String? = null

        companion object {

            fun objectFromData(str: String): ResultBean {

                return Gson().fromJson(str, ResultBean::class.java)
            }

            fun objectFromData(str: String, key: String): ResultBean? {

                try {
                    val jsonObject = JSONObject(str)

                    return Gson().fromJson(jsonObject.getString(str), ResultBean::class.java)
                } catch (e: JSONException) {
                    e.printStackTrace()
                }

                return null
            }

            fun arrayResultBeanFromData(str: String): List<ResultBean>? {

                val listType = object : TypeToken<ArrayList<ResultBean>>() {

                }.type

                return Gson().fromJson<List<ResultBean>>(str, listType)
            }

            fun arrayResultBeanFromData(str: String, key: String): List<ResultBean>? {

                try {
                    val jsonObject = JSONObject(str)
                    val listType = object : TypeToken<ArrayList<ResultBean>>() {

                    }.type

                    return Gson().fromJson<List<ResultBean>>(jsonObject.getString(str), listType)

                } catch (e: JSONException) {
                    e.printStackTrace()
                }

                return ArrayList()


            }
        }
    }

    companion object {

        fun objectFromData(str: String): ZhouGongBean {

            return Gson().fromJson(str, ZhouGongBean::class.java)
        }

        fun objectFromData(str: String, key: String): ZhouGongBean? {

            try {
                val jsonObject = JSONObject(str)

                return Gson().fromJson(jsonObject.getString(str), ZhouGongBean::class.java)
            } catch (e: JSONException) {
                e.printStackTrace()
            }

            return null
        }

        fun arrayZhouGongBeanFromData(str: String): List<ZhouGongBean>? {

            val listType = object : TypeToken<ArrayList<ZhouGongBean>>() {

            }.type

            return Gson().fromJson<List<ZhouGongBean>>(str, listType)
        }

        fun arrayZhouGongBeanFromData(str: String, key: String): List<ZhouGongBean>? {

            try {
                val jsonObject = JSONObject(str)
                val listType = object : TypeToken<ArrayList<ZhouGongBean>>() {

                }.type

                return Gson().fromJson<List<ZhouGongBean>>(jsonObject.getString(str), listType)

            } catch (e: JSONException) {
                e.printStackTrace()
            }

            return ArrayList()


        }
    }
}
